libs/api/domains/hms-loans/src/lib/api-domains-hms-loans.resolver.ts

```diff
@@ -0,0 +1,36 @@
+import { Query, Resolver } from '@nestjs/graphql'
+import { UseGuards } from '@nestjs/common'
+
+import { ApiScope } from '@island.is/auth/scopes'
+import type { User } from '@island.is/auth-nest-tools'
+import {
+  IdsUserGuard,
+  ScopesGuard,
+  CurrentUser,
+  Scopes,
+} from '@island.is/auth-nest-tools'
+import { HmsLoansClientService } from '@island.is/clients/hms-loans'
+import { Audit } from '@island.is/nest/audit'
+
+import { LoanOverview } from './models/loanOverview.model'
+import { PaymentOverview } from './models/paymentOverview.model'
+
+@UseGuards(IdsUserGuard, ScopesGuard)
+@Scopes(ApiScope.financeLoans)
+@Resolver()
+@Audit({ namespace: '@island.is/api/hms-loans' })
+export class HmsLoansResolver {
+  constructor(private hmsLoansService: HmsLoansClientService) {}
+
+  @Query(() => [LoanOverview], { nullable: true })
+  @Audit()
+  async getHmsLoansLoanOverview(@CurrentUser() user: User) {
+    return await this.hmsLoansService.getHmsLoansLoanOverview(user)
+  }
+
+  @Query(() => [PaymentOverview], { nullable: true })
+  @Audit()
+  async getHmsLoansPaymentOverview(@CurrentUser() user: User) {
+    return this.hmsLoansService.getHmsLoansPaymentOverview(user)
+  }
```

Comment:
The [naming convention](https://github.com/island-is/island.is/pull/12730) of queries would be `hmsLoansLoanOverview` and `hmsLoansPaymentOverview` where they start with they domain name `hmsLoans` then a noun for the object/collection being queried. But I would also consider shortening them to `hmsLoans` for the list of loans (we don't need to repeat `Loans` when the object name is in the domain name) and `hmsLoansPayments`

```suggestion
  @Query(() => [LoanOverview], { name: 'hmsLoans', nullable: true })
  @Audit()
  async getHmsLoans(@CurrentUser() user: User) {
    return await this.hmsLoansService.getHmsLoansLoanOverview(user)
  }

  @Query(() => [PaymentOverview], { name: 'hmsLoansPayments', nullable: true })
  @Audit()
  async getHmsLoansPayments(@CurrentUser() user: User) {
    return this.hmsLoansService.getHmsLoansPaymentOverview(user)
  }
```